### PR TITLE
Thin instances: Automatic buffer recreation when static buffer changes

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/unable-to-set-thin-instance-with-thininstancesetmatrixat/48896/11

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14901/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14901/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14901/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14901/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14901/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14901/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14901/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14901/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14901/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14901/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Meshes/thinInstanceMesh.ts [hunk 1]
```diff
@@ -11,6 +11,15 @@ declare module "./mesh" {
          * Gets or sets a boolean defining if we want picking to pick thin instances as well
          */
         thinInstanceEnablePicking: boolean;
+
+        /**
+         * Indicates that a buffer created as static should be recreated if the buffer is updated (by calling thinInstanceSetMatrixAt or thinInstanceSetAttributeAt, for eg.)
+         * If this flag is false, a buffer created as "static" won't show any update done to it, and will stay the same as it was created.
```
Comment (RaananW) [inline packages/dev/core/src/Meshes/thinInstanceMesh.ts:17]:
```suggestion
         * If this flag is false (the default behavior), a buffer created as "static" won't show any update done to it, and will stay the same as it was created.
```

Nit, feel free to ignore that

**Review by RaananW — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14901/merge/testResults/webgpuplaywright/index.html

**Pull request merged**